test/fuzz/luaL_loadbuffer/serializer.cc

```diff
@@ -5,6 +5,147 @@
  */
 #include "serializer.h"
 
+#include <stack>
+#include <string>
+
+namespace {
+
+const std::string kCounterNamePrefix = "counter_";
+
+/**
+ * Class that controls id creation for counters. Basically, a variable wrapper
+ * that guarantees variable to be incremented.
+ */
+class CounterIdProvider_ {
+public:
+	/** Returns number of id provided. */
+	std::size_t
+	count()
+	{
+		return id_;
+	}
+
+	/** Returns a new id that was not used after last clean(). */
+	std::size_t
+	next()
+	{
+		return id_++;
+	}
+
+	/** Cleans history. Should be used to make fuzzer starts independent. */
+	void
+	clean()
+	{
+		id_ = 0;
+	}
+
+private:
+	std::size_t id_ = 0;
+};
+
+/** A singleton for counter id provider. */
+CounterIdProvider_&
+counter_id_provider()
+{
+	static CounterIdProvider_ provider;
+	return provider;
+}
+
+std::string
+get_counter_name(std::size_t id)
+{
+	return kCounterNamePrefix + std::to_string(id);
+}
+
+/** Returns `<counter_name> = <counter_name> + 1`. */
+std::string
+get_counter_increment(const std::string &counter_name)
+{
+	std::string retval = counter_name;
+	retval += " = ";
+	retval += counter_name;
+	retval += " + 1\n";
+	return retval;
+}
+
+/** Returns `if <counter_name> > kMaxCounterValue then <then_block> end`. */
+std::string
+get_condition(const std::string &counter_name, const std::string &then_block)
+{
+	std::string retval = "if ";
+	retval += counter_name;
+	retval += " > ";
+	retval += std::to_string(kMaxCounterValue);
+	retval += " then ";
+	retval += then_block;
+	retval += " end\n";
+	return retval;
+}
+
+/**
+ * Block may be placed not only in a cycle, so specially for cycles there is
+ * a function that will add break condition and counter increment.
+ */
+std::string
+BlockToStringCycleProtected(const Block &block, const std::string &counter_name)
+{
+	std::string retval = get_condition(counter_name, "break");
+	retval += get_counter_increment(counter_name);
```

Comment:
I see that this token ("break") is used twice in the code base, I suggest to create a constant (named `TokBreak` for example) to use it in all such places.
Same for "return".

Reply:
I am not sure if there is any point to it, as they were used in different contexts. Additionally, in my next PR, I will remove the instances of `"return"` and `"break"` that were not placed by me.